v3_async_wip/v3_async_wip/iothub_session.py

```diff
@@ -234,9 +254,15 @@ async def direct_method_requests(
         """Returns an async generator of incoming direct method requests"""
         await self._mqtt_client.enable_direct_method_request_receive()
         try:
-            yield self._mqtt_client.incoming_direct_method_requests
+            yield self._add_disconnect_interrupt_to_generator(
+                self._mqtt_client.incoming_direct_method_requests
+            )
         finally:
-            await self._mqtt_client.disable_direct_method_request_receive()
+            try:
+                await self._mqtt_client.disable_direct_method_request_receive()
+            except mqtt.MQTTError:
+                # i.e. not connected
```

Comment:
># i.e. not connected [](http://example.com/codeflow?start=16&length=20)

It might be better to say something like "OK to ignore. We expect MQTTError to be raised here after disconnection." or something like that.